### ISSUE TITLE
Varlociraptor preprocess fails for input without records

### ISSUE THREAD

**Issue by FelixMoelder** [varlociraptor/varlociraptor#110]
When varlociraptor receives an bcf-file without any records preprocessing fails with the following message:

```
Found no records to use for estimating the insert size. Will assume
single end sequencing data and calculate deletion probabilities without
considering the insert size.

If your data should be paired end, please consider manually providing
--alignment-properties, e.g. computed with `samtools stats`. Also,
the following counts of unusable records might indicate a source of
this problem:

- I, D, S or H CIGAR operation: 9591
- S CIGAR (soft clip, e.g. due to UMIs or adapters): 9591

In addition, 10409 records were skipped in the estimation for one
of the following reasons:
- not paired
- not the first segment with regard to the template sequence
- mapping quality of 0
- marked as a duplicate
- mate mapped to different template (e.g. different chromosome)
- failed some quality check according to the 512 SAM flag
- mate unmapped
- record unmapped

thread 'main' panicked at 'This is a bug.

                            AlignmentProperties has insert_size set to `None`, but you are trying to use it without checking for None.', src/estimation/alignment_properties.rs:204:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

While this case seems to be expected it is not handled well, as I would expect to receive an empty output file.

**Comment by dlaehnemann** [comment]
The message refers to the BAM file and the insert size estimation there. So I'm not sure that this is related to the empty BCF file, although that might be responsible for triggering a code path where the behaviour described by the `panic!` happens. Could you rerun with `RUST_BACKTRACE=1` and post the output here? I'm hoping this will give some more info on the code path where this happens and help isolate the bug. It then shouldn't be too difficult to squash.

**Comment by johanneskoester** [comment]
Indeed, the issue was the BAM. Namely, due to primer trimming it contained almost entirely hardclipped reads. Those were discarded before. With the release I will publish today, there is a flag for disabling insert size based evidence (which makes no sense on amplicon data) which also circumvents this issue. Nevertheless, what remains is the problem that when there are not enough candidates in normal mode, we reach the None issue at the bottom. Just assuming that the data is single end is probably not a good idea. The fix will be to fail instead of warn, and only proceed if a flag --single-end or so was specified.

Hence leaving this open.

**Comment by dlaehnemann** [comment]
Yep, failing instead of warning is a good idea, if you then suggest a single-end flag right away in the error message.

**Comment by jackwadden** [comment]
I'm also seeing this issue. Is there a fix or example usage to properly call variants from amplicon data?
